### PR TITLE
soa non personal with out confidential checks

### PR DESCRIPTION
**Pull request by sahitya-desireddy**
### JIRA link (if applicable) ###
https://tools.hmcts.net/jira/browse/PRL-5224

### Change description ###
SOA - Pick post contact preference --CA Non personal service - Applicant LIP (citizen initiated) Part 1 No confidentiality check event needed

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: src/main/java/uk/gov/hmcts/reform/prl/services/ServiceOfApplicationService.java [hunk 1]
```diff
@@ -1272,12 +1302,16 @@ private List<Element<EmailNotificationDetails>> sendNotificationsToCitizenRespon
...
                     log.info("Respondent is represented");
                     try {
-                        emailNotificationDetails.add(element(serviceOfApplicationEmailService.sendEmailNotificationToSolicitor(
-                            authorization, caseData,
-                            selectedRespondent.getValue(),
-                            isStaticDocs ? getNotificationPack(caseData, PrlAppsConstants.S, docs) : docs,
-                            SERVED_PARTY_RESPONDENT
-                        )));
+                        Map<String, Object> dynamicData = EmailUtils.getCommonSendgridDynamicTemplateData(caseData);
+                        dynamicData.put("name", selectedRespondent.getValue().getRepresentativeFullName());
```
Comment (shashisk97) [inline src/main/java/uk/gov/hmcts/reform/prl/services/ServiceOfApplicationService.java:1306]: casename and reference already set in utils method

File: src/main/java/uk/gov/hmcts/reform/prl/services/ServiceOfApplicationService.java [hunk 2]
```diff
@@ -1272,12 +1302,16 @@ private List<Element<EmailNotificationDetails>> sendNotificationsToCitizenRespon
...
+                        dynamicData.put("caseName", caseData.getApplicantCaseName());
+                        dynamicData.put("caseNumber", caseData.getId());
+                        dynamicData.put(DASH_BOARD_LINK, manageCaseUrl + PrlAppsConstants.URL_STRING + caseData.getId());
```
Comment (shashisk97) [inline src/main/java/uk/gov/hmcts/reform/prl/services/ServiceOfApplicationService.java:1309]: What should be the url here? citizen or manage case?

File: src/main/java/uk/gov/hmcts/reform/prl/services/ServiceOfApplicationService.java [hunk 3]
```diff
@@ -1448,8 +1482,20 @@ private void sendPostWithAccessCodeLetterToParty(CaseData caseData, String autho
                                                      Element<PartyDetails> party, String template,
                                                      String servedParty) {
 
+
+        sendPostWithAccessCodeLetterToParty(caseData,authorization,packDocs,bulkPrintDetails,party,template,servedParty,true);
```
Comment (shashisk97) [inline src/main/java/uk/gov/hmcts/reform/prl/services/ServiceOfApplicationService.java:1486]: Can you please check with Sairam as it has been implmeneted and this refactoring is not needed.

File: src/main/java/uk/gov/hmcts/reform/prl/services/ServiceOfApplicationService.java (hunk 1 above)
Comment (sahitya-desireddy) [reply src/main/java/uk/gov/hmcts/reform/prl/services/ServiceOfApplicationService.java:1306]: addressed this Thank you !!

File: src/main/java/uk/gov/hmcts/reform/prl/services/ServiceOfApplicationService.java [hunk 4]
```diff
@@ -1226,25 +1233,35 @@ private List<Element<EmailNotificationDetails>> sendNotificationsToCitizenApplic
                     caseInvite = c100CaseInviteService.generateCaseInvite(selectedApplicant, Yes);
                     caseInvites.add(element(caseInvite));
                 }
+                Map<String, Object> dynamicData = EmailUtils.getCommonSendgridDynamicTemplateData(caseData);
+                dynamicData.put("name", selectedApplicant.getValue().getFirstName() + " "
+                    + selectedApplicant.getValue().getLastName());
+                dynamicData.put(DASH_BOARD_LINK, manageCaseUrl + PrlAppsConstants.URL_STRING + caseData.getId());
```
[the remark this answers is not in the document]
Comment (sahitya-desireddy) [reply src/main/java/uk/gov/hmcts/reform/prl/services/ServiceOfApplicationService.java]: Thanks @shashisk97 changed it to citizen url please ...

File: src/main/java/uk/gov/hmcts/reform/prl/services/ServiceOfApplicationService.java [hunk 5]
```diff
@@ -1272,12 +1302,16 @@ private List<Element<EmailNotificationDetails>> sendNotificationsToCitizenRespon
...
-                            SERVED_PARTY_RESPONDENT
-                        )));
+                        Map<String, Object> dynamicData = EmailUtils.getCommonSendgridDynamicTemplateData(caseData);
+                        dynamicData.put("name", selectedRespondent.getValue().getRepresentativeFullName());
+                        dynamicData.put("caseName", caseData.getApplicantCaseName());
+                        dynamicData.put("caseNumber", caseData.getId());
+                        dynamicData.put(DASH_BOARD_LINK, manageCaseUrl + PrlAppsConstants.URL_STRING + caseData.getId());
+                        emailNotificationDetails.add(element(serviceOfApplicationEmailService
+                                                                 .sendEmailUsingTemplateWithAttachments(authorization,
+                                                selectedRespondent.getValue().getEmail(), docs,
```
[the remark this answers is not in the document]
Comment (sahitya-desireddy) [reply src/main/java/uk/gov/hmcts/reform/prl/services/ServiceOfApplicationService.java]: yes changed to solicitor email kindly review it Thank you @shashisk97

File: src/main/java/uk/gov/hmcts/reform/prl/services/ServiceOfApplicationService.java (hunk 3 above)
Comment (sahitya-desireddy) [reply src/main/java/uk/gov/hmcts/reform/prl/services/ServiceOfApplicationService.java]: pushed the changes request to please kindly review it Thanks @shashisk97

**Pull request closed without merging**